### PR TITLE
greendns: Use _compute_times on dnspython 2 (Updated)

### PR DESCRIPTION
**Pull request by felixonmars**
This PR is based on https://github.com/eventlet/eventlet/pull/639

Rebased against latest master, and updated the GitHub CI instead of Travis.

I have also updated the tests for the last test failure left.

### CODE REVIEW

File: eventlet/support/greendns.py
Comment (temoto) [inline eventlet/support/greendns.py:126]: Think you can do this attribute check at module level to avoid runtime cost?

File: tox.ini
Comment (temoto) [inline tox.ini:19]:
```suggestion
    dnspython1, ipv6, pep8, py{27,35,36,37,38,39,py2,py3}-{selects,poll,epolls}, py38-openssl
```

IMHO, it's sufficient to check against dnspython1 on any single Python version.

File: .github/workflows/test.yaml
Comment (temoto) [inline .github/workflows/test.yaml:56]:
```suggestion
          - { py: 3.x, toxenv: dnspython1, ignore-error: false }
```
+ remove combination explosion above

IMHO, it's sufficient to check against dnspython1 on any single Python version.

Comment (felixonmars) [reply .github/workflows/test.yaml:52]: Should I literally write `3.x` here? (Sorry I'm not so familiar)

Otherwise I have updated to only test on Python 2.7 & 3.9.

Comment (temoto) [reply .github/workflows/test.yaml:52]: Yeah 27 is fine even if redundant. Also you can press "commit suggestion" button if you agree with proposed code.

Comment (temoto) [reply .github/workflows/test.yaml:52]: 3.x selects "latest of 3.any"

Comment (felixonmars) [reply .github/workflows/test.yaml:52]: I see. I thought 3.x was invalid so I didn't click the button. Sorry :P

File: eventlet/support/greendns.py
Comment (felixonmars) [reply eventlet/support/greendns.py]: Updated too :)

**Comment by felixonmars** [comment]
Okay, found the cause. They are checking whether the QR flag is set now.

All green now :)

**Review by temoto — APPROVED** [review]
LGTM, **but** I'm pretty sure someone reported that compute_time wasn't enough for compatibility with dnspython2.

So to actually merge this, we need to run some real world application using updated code. Testing volunteers are very welcome.

**Comment by felixonmars** [comment]
> LGTM, **but** I'm pretty sure someone reported that compute_time wasn't enough for compatibility with dnspython2.
> 
> So to actually merge this, we need to run some real world application using updated code. Testing volunteers are very welcome.

I took a look at the other related issue: #632

The original logs are not accessible now, but based on on nova's bug report this seems to be caused by a module import order issue: https://bugs.launchpad.net/nova/+bug/1888237

In #684 we are patching ssl now, not sure if it helps on this. But IMHO if it's indeed caused by a module import order issue, it should be fixed at the corresponding project (by running eventlet monkeypatch in prior to importing dnspython), and we can do very little here.

**Comment by temoto** [comment]
Maintainer reminder before merge: add co-authored-by @ralonsoh and @jayvdb

**Comment by rthalley** [comment]
See my comments in #619 [here](https://github.com/eventlet/eventlet/issues/619#issuecomment-660250478).  There were more changes than to just compute_times, as the API of dns.query.udp() and dns.query.tcp() changed too, and eventlet monkeypatching them back to old versions will cause failures with the resolver.

**Comment by felixonmars** [comment]
@rthalley Thank you, and sorry for missing that comment before. I have added all the missing parameters for udp() and tcp(), up to master. Hopefully the implementation makes sense.

**Comment by filippog** [comment]
I'm using python-eventlet from Debian Bullseye (`0.26.1-7`) and can confirm that with these changes DNS resolution works again with dnspython `2.0.0` and Openstack Swift `2.26.0-10`

**Pull request merged**
